handler/api/src/main/java/io/knotx/fragments/handler/api/actionlog/ActionLogger.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.handler.api.actionlog;
+
+import static io.knotx.fragments.handler.api.actionlog.ActionLogMode.INFO;
+
+import java.time.Instant;
+import java.util.function.Function;
+
+import io.knotx.fragments.handler.api.ActionConfig;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class ActionLogger {
+
+  private final ActionLogMode actionLogMode;
+  private final JsonObject logs;
+  private final JsonObject actionLog;
+
+  private ActionLogger(String alias, ActionLogMode actionLogMode) {
+    this.actionLogMode = actionLogMode;
+    this.logs = new JsonObject();
+    this.actionLog = new JsonObject().put("alias", alias)
```

Comment:
Can we introduce some model here:
```
class ActionLog {
  String alias;
  JsonObject logs;
  ActionLog[] doAction;
}
```
This structure is common for all actions. Only the `logs` variable is a JSON object that contains action specific details. Please remember to implement `toString()`, `hashCode()` and `equals()` methods.